pkg/util/runnable.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+)
+
+// Runnable is for handling sub components in a unified way.
+//
+// For example:
+// r := createRunable()
+// ctx, cancel := context.WithCancel(context.Background())
+// go func() { handleError(r.Run(ctx)) }()
+// r.WaitForReady()
+// cancel()
+// r.Close()
+type Runnable interface {
+	// Run all sub goroutines and block the current one. If an error occurs
+	// in any sub goroutine, return it and cancel all others.
+	Run(ctx context.Context) error
```

Comment:
Whether the blocked Run releases resources before exiting? If so, is the Close method redundant?